### PR TITLE
Implement contract status filter

### PR DESCRIPTION
**Pull request by Aditya-A-G**
Screenshot from 2023-05-23 01-59-57
Screenshot from 2023-05-23 02-00-29

### CODE REVIEW

File: src/components/ContractsSidebar/index.jsx [hunk 1]
```diff
@@ -1,10 +1,66 @@
...
+  'delivered',
+  'approved',
+  'declined',
+  'withdrew',
+  'refunded',
+  'closed',
+]
+
+export const ContractsSidebar = ({
+  // as,
```
Comment (gruz0) [inline src/components/ContractsSidebar/index.jsx:24]: We can remove this line

File: src/components/ContractsSidebar/index.jsx [hunk 2]
```diff
@@ -18,11 +74,12 @@ export const ContractsSidebar = ({ as }) => {
...
             label={`${t('components.contracts_sidebar.status')}:`}
-            placeholder={t('components.contracts_sidebar.all')}
-            disabled
+            options={statusOptions()}
+            value={filters.status}
+            onChange={handleStatusChange}
           />
 
-          <Form.Select
+          {/* <Form.Select
```
Comment (gruz0) [inline src/components/ContractsSidebar/index.jsx:82]: Please remove all commented lines also

File: src/hooks/useContractsFilter.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import { useState, useEffect } from 'react'
+
+export const useContractsFilter = ({ data, filters }) => {
+  const [filteredJobs, setFilteredJobs] = useState([])
```
Comment (gruz0) [inline src/hooks/useContractsFilter.js:4]: Why `jobs`? Please pay attention, don't copy/paste blindly.

File: src/hooks/useContractsFilter.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+  const filterByStatus = (status = 'active', filtered) => {
+    if (status === 'all') return filtered
+
+    if (status === 'active') {
+      return filtered.filter((job) => job.status !== 'closed')
```
Comment (gruz0) [inline src/hooks/useContractsFilter.js:10]: `job` instead of `contract`

File: src/hooks/useContractsFilter.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    }
+
+    return filtered.filter((job) => job.status === status)
```
Comment (gruz0) [inline src/hooks/useContractsFilter.js:13]: `job` instead of `contract`

File: src/hooks/useContractsFilter.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+  }
+
+  useEffect(() => {
+    if (!data) return
+
+    let filtered = data
+
+    filtered = filterByStatus(filters.status, filtered)
+
+    setFilteredJobs(filtered)
```
Comment (gruz0) [inline src/hooks/useContractsFilter.js:23]: `job` instead of `contract`

File: src/hooks/useContractsFilter.js [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+  }, [data, filters])
+
+  return filteredJobs
```
Comment (gruz0) [inline src/hooks/useContractsFilter.js:26]: `job` instead of `contract`

**Review by gruz0 — CHANGES_REQUESTED** [review]

File: src/hooks/useContractsFilter.js (hunk 3 above)
Comment (Aditya-A-G) [reply src/hooks/useContractsFilter.js:4]: My apologies for the oversight. forgot to adjust the naming. Thank you for catching that.

**Review by gruz0 — APPROVED** [review]

**Pull request merged**